timetracker/entries/views.py

```diff
@@ -1,10 +1,46 @@
 from django.shortcuts import render
 
-from .models import Entry
+from .models import Entry, Client, Project
 
 
 def entries(request):
     entry_list = Entry.objects.all()
     return render(request, 'entries.html', {
         'entry_list': entry_list,
+
+    })
+
+def projects(request):
+    project_list = Project.objects.all()
+    return render(request, 'projects.html', {
+        'project_list': project_list,
     })
+
+def clients(request):
+    client_list = Client.objects.all()
+    return render(request, 'clients.html', {
+        'client_list': client_list,
+    })
+
+def client_summary(request, pk):
+    client = Client.objects.get(pk=pk)
+    projects = client.project_set.all()
+    
+    # project_summaries = []
+    project_summaries = [project.entry_set.all() for project in projects]
```

Comment:
You don't necessarily fetch all the objects here, but do that in the template (see comment above)